pkg/apis/tuned/v1/tuned_types.go

```diff
@@ -144,13 +146,65 @@ type ProfileStatus struct {
 	// kernel parameters calculated by tuned for the active Tuned profile
 	// +optional
 	Bootcmdline string `json:"bootcmdline"`
-	// deploy stall daemon: https://github.com/bristot/stalld/
+
+	// the current profile in use by the Tuned daemon
+	TunedProfile string `json:"tunedProfile"`
+
+	// conditions represents the state of the per-node Profile application
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +optional
+	Conditions []ProfileStatusCondition `json:"conditions,omitempty"  patchStrategy:"merge" patchMergeKey:"type"`
+
+	// deploy stall daemon: https://git.kernel.org/pub/scm/utils/stalld/stalld.git
 	// +optional
 	Stalld bool `json:"stalld"`
 }
 
+// ProfileStatusCondition represents a partial state of the per-node Profile application.
+// +k8s:deepcopy-gen=true
+type ProfileStatusCondition struct {
+	// type specifies the aspect reported by this condition.
+	// +kubebuilder:validation:Required
+	// +required
+	Type ProfileConditionType `json:"type"`
+
+	// status of the condition, one of True, False, Unknown.
+	// +kubebuilder:validation:Required
+	// +required
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastTransitionTime is the time of the last update to the current status property.
+	// +kubebuilder:validation:Required
+	// +required
+	LastTransitionTime metav1.Time `json:"lastTransitionTime"`
+
+	// reason is the CamelCase reason for the condition's current status.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message provides additional information about the current condition.
+	// This is only to be consumed by humans.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
```

Comment:
@jmencak looks good, I think we better merge it and consume it with PAO, and then we can go from there. We will have a better visibility and we can come up with improvement ideas, if needed.

Reply:
Thanks, @marcel-apf .  I'll aim to merge this immediately once 4.8 opens.  @sjug/@dagrayvid , could you please provide a code review?  Thank you!